src/main/java/com/epam/university/java/core/task006/Task006Impl.java

```diff
@@ -0,0 +1,47 @@
+package com.epam.university.java.core.task006;
+
+import java.util.Collection;
+import java.util.function.ToDoubleFunction;
+
+import com.epam.university.java.core.task003.NullChecker;
+import com.epam.university.java.core.task003.SimpleNullChecker;
+/**
+ * Created by ilya on 06.09.17.
+ */
+public class Task006Impl implements Task006 {
+    private Collection<Measurement> measurements;
```

Comment:
You shouldn't store state. In the case of parallel execution can be race conditions